src/FileMaxSizeReachedException.java

```diff
@@ -0,0 +1,6 @@
+public class FileMaxSizeReachedException extends Exception {
```

Comment:
For the future, you should extend your custom exceptions from `RuntimeException`, not from `Exception`